x-pack/legacy/plugins/lens/public/editor_frame_plugin/editor_frame/config_panel_header.tsx

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState, useMemo } from 'react';
+import {
+  EuiIcon,
+  EuiPopover,
+  EuiButton,
+  EuiPopoverTitle,
+  EuiKeyPadMenu,
+  EuiKeyPadMenuItemButton,
+  EuiOverlayMask,
+  EuiConfirmModal,
+} from '@elastic/eui';
+import { flatten } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { NativeRenderer } from '../../native_renderer';
+import { Visualization, FramePublicAPI } from '../../types';
+import { Action } from './state_management';
+
+interface VisualizationSelection {
+  visualizationId: string;
+  subVisualizationId: string;
+}
+
+interface State {
+  isFlyoutOpen: boolean;
+  confirmVisualization?: VisualizationSelection;
+}
+
+interface Props {
+  dispatch: (action: Action) => void;
+  visualizationMap: Record<string, Visualization>;
+  visualizationId: string | null;
+  visualizationState: unknown;
+  framePublicAPI: FramePublicAPI;
+}
+
+function getSuggestion(frame: FramePublicAPI, visualization: Visualization) {
+  const layers = Object.entries(frame.datasourceLayers);
+
+  if (!layers.length) {
+    return {
+      suggestion: undefined,
+      layerId: undefined,
+    };
+  }
+
+  const [[layerId, datasource]] = layers;
+  const suggestions = visualization.getSuggestions({
+    tables: [
+      {
+        datasourceSuggestionId: 0,
+        isMultiRow: true,
+        columns: datasource.getTableSpec().map(col => ({
+          ...col,
+          operation: datasource.getOperationForColumnId(col.columnId)!,
+        })),
+        layerId,
+      },
+    ],
+  });
+
+  const suggestion = suggestions.length ? suggestions[0] : undefined;
```

Comment:
This logic is too simplistic- I would expect that if there are multiple possible suggestions, the user might be prompted to choose one?

Reply:
You are right, I changed this to list the possible resulting charts if there are multiple suggestions.
<img width="942" alt="Screenshot 2019-07-30 at 11 24 46" src="https://user-images.githubusercontent.com/1508364/62118174-83b8cc00-b2bd-11e9-93f5-f3d1d1699451.png">

What do you think?